lib/overcommit/utils.rb

```diff
@@ -59,7 +59,7 @@ def repo_root
       def git_dir
         @git_dir ||=
           begin
-            result = execute(%w[git rev-parse --git-common-dir])
+            result = execute(%w[git rev-parse --git-dir])
```

Comment:
This breaks support for [`git worktree`](https://git-scm.com/docs/git-worktree) (shame on me for not adding a proper test explicitly using worktrees in 10c20808).

In the meantime, as long we are running Git 2.5+ we should use `--git-common-dir`, otherwise we can use `--git-dir`. You can express this as:

```ruby
cmd = %w[git rev-parse]
cmd << GIT_VERSION < '2.5' ? '--git-dir' : '--git-common-dir'
result = execute(cmd)
```